StitchCore/StitchCoreTests/Admin/StitchAdminClient.swift

```diff
@@ -0,0 +1,186 @@
+        }
+    }
+}
+
+// Adds an endpoint method that DELETEs some id
+internal protocol Remove: Endpoint { }
+extension Remove {
+    public func remove() -> Promise<Any> {
+        return self.httpClient.doRequest {
+            $0.endpoint = self.url
+            $0.method = .delete
+        }
+    }
+}
+
+// Adds an endpoint method that POSTs new data
+internal protocol Create: Endpoint {
+    associatedtype CreatorModel: Encodable
+    associatedtype Model: Decodable
+}
+extension Create {
+    public func create(data: CreatorModel) -> Promise<Model> {
+        return self.httpClient.doRequest {
+            $0.endpoint = self.url
+            $0.method = .post
+            try $0.encode(withData: data)
+        }.flatMap {
+            return try JSONDecoder().decode(Model.self,
+                                            from: JSONSerialization.data(withJSONObject: $0))
+        }
+    }
+}
+
+// Adds an endpoint method that PUTs some data
+internal protocol Update: Endpoint { associatedtype Model: Codable }
+extension Update {
+    public func update(data: Model) -> Promise<Model> {
+        return self.httpClient.doRequest {
+            $0.endpoint = self.url
+            $0.method = .put
+            try $0.encode(withData: data)
+        }.flatMap {
+            return try JSONDecoder().decode(Model.self,
+                                            from: JSONSerialization.data(withJSONObject: $0))
+        }
+    }
+}
+
+// Adds an endpoint that enables a given resource
+internal protocol Enable: Endpoint { }
+extension Enable {
+    public func enable() -> Promise<Any> {
+        return self.httpClient.doRequest {
+            $0.endpoint = "\(self.url)/enable"
+            $0.method = .put
+        }
+    }
+}
+
+// Adds an endpoint that disables a given resource
+internal protocol Disable: Endpoint { }
+extension Disable {
+    public func disable() -> Promise<Any> {
+        return self.httpClient.doRequest {
+            $0.endpoint = "\(self.url)/disable"
+            $0.method = .put
+        }
+    }
+}
+
+public final class StitchAdminClientFactory {
+    public static func create(baseUrl: String = Consts.defaultServerUrl) -> Promise<StitchAdminClient> {
+        return Promise(value: StitchAdminClient.init(baseUrl: baseUrl))
+    }
+}
+
+public class StitchAdminClient {
+    let baseUrl: String
+    let httpClient: StitchHTTPClient
+
+    fileprivate init(baseUrl: String) {
+        self.baseUrl = baseUrl
+        self.httpClient = StitchHTTPClient.init(baseUrl: baseUrl,
+                                                apiPath: "/api/admin/v3.0",
+                                                networkAdapter: StitchNetworkAdapter(),
+                                                storage: MemoryStorage(),
+                                                storageKeys: StorageKeys.init(suiteName: "__admin__"))
+    }
+
+    func apps(withGroupId groupId: String) -> AppsEndpoint {
```

Comment:
[opinion] From an API perspective, receiving an "endpoint" doesn't feel right. Why don't we break from what the JS SDK did and not offer a hierarchy of "endpoints"? I think it makes the world confusing versus just `adminClient.getApp(:withAppId:inGroupId)` or something. That would significantly decrease the complexity and remove the need for endpoints and many protocols as well as decouple httpclients from endpoints. You could build each of these methods as grouped extensions in a different file.